packages/components/src/base-field/hook.js

```diff
@@ -0,0 +1,74 @@
+/**
+ * External dependencies
+ */
+import { cx } from 'emotion';
+
+/**
+ * WordPress dependencies
+ */
+import { useMemo } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import { useContextSystem } from '../ui/context';
+import { useControlGroupContext } from '../ui/control-group';
+import { useFlex } from '../flex';
+import * as styles from './styles';
+
+/**
+ * @typedef OwnProps
+ * @property {boolean} [error=false] Renders an error.
+ * @property {boolean} [disabled] Whether the field is disabled.
+ * @property {boolean} [isClickable=false] Renders a `cursor: pointer` on hover.
+ * @property {boolean} [isFocused=false] Renders focus styles.
+ * @property {boolean} [isInline=false] Renders as an inline element (layout).
+ * @property {boolean} [isSubtle=false] Renders a subtle variant.
+ */
+
+/** @typedef {import('../Flex/useFlex').FlexProps & OwnProps} Props */
+
+/**
+ * @param {import('@wp-g2/create-styles').ViewOwnProps<Props, 'div'>} props
+ */
+export function useBaseField( props ) {
+	const {
+		className,
+		error = false,
```

Comment:
1. Since this prop is a `boolean`, should it be called `isError`?
2. In case a custom error message needs to be shown, is that something that will be handled by the component extending `BaseField`?

Reply:
`isError` doesn't quite read as semantic english to me. `hasError` would be better if we wanted to give it a verb.

I think the error message itself would be displayed by the field rather than `base-field` 🤔 Especially considering `BaseField` is only used as a hook in the places it's used (https://github.com/ItsJonQ/g2/blob/main/packages/components/src/TextInput/useTextInput.js#L120) I don't think `BaseField` can actually render anything.

Maybe in that case it shouldn't even be a component and should just be a hook. What do you think?

Reply:
> `hasError` would be better

Yup, agreed!

> Maybe in that case it shouldn't even be a component and should just be a hook. What do you think?

This also makes sense. We can have only as a hook for now, and if we ever have the need, create the component version too